### ISSUE TITLE
Message_AmqpConnectionLossSendRecovery_AmqpWs failed with Microsoft.ServiceBus.Messaging.MessagingException: 

### ISSUE THREAD

**Issue by CIPop** [Azure/azure-iot-sdk-csharp#230]
Failed during CI testing in #224.

The exception is suggesting retry.

```
Test method Microsoft.Azure.Devices.E2ETests.MessageE2ETests.Message_AmqpConnectionLossSendRecovery_AmqpWs threw exception: 
Microsoft.ServiceBus.Messaging.MessagingException: The service was unable to process the request; please retry the operation. For more information on exception types and proper exception handling, please refer to http://go.microsoft.com/fwlink/?LinkId=761101 TrackingId:bc9a792b57ab4d2499a8cd0a03ee4ad6_G6, SystemTracker:gateway6, Timestamp:11/15/2017 2:58:54 AM, referenceId: bc9a792b57ab4d2499a8cd0a03ee4ad6_G6```

```
Microsoft.ServiceBus.Common.AsyncResult.End[TAsyncResult](IAsyncResult result)
Microsoft.ServiceBus.Messaging.Amqp.AmqpMessagingFactory.EndOpenEntity(IAsyncResult result)
Microsoft.ServiceBus.Messaging.Amqp.AmqpMessageReceiver.EndCreateLink(IAsyncResult result)
Microsoft.ServiceBus.Messaging.Amqp.FaultTolerantObject`1.CreateAsyncResult.<>c.<GetAsyncSteps>b__5_1(CreateAsyncResult thisPtr, IAsyncResult r)
Microsoft.ServiceBus.Messaging.IteratorAsyncResult`1.StepCallback(IAsyncResult result)
Microsoft.ServiceBus.Common.AsyncResult.End[TAsyncResult](IAsyncResult result)
Microsoft.ServiceBus.Common.AsyncResult`1.End(IAsyncResult asyncResult)
Microsoft.ServiceBus.Messaging.Amqp.FaultTolerantObject`1.OnEndCreateInstance(IAsyncResult asyncResult)
Microsoft.ServiceBus.Messaging.SingletonManager`1.EndGetInstance(IAsyncResult asyncResult)
Microsoft.ServiceBus.Messaging.Amqp.AmqpMessageReceiver.OnEndOpen(IAsyncResult result)
Microsoft.ServiceBus.Messaging.ClientEntity.EndOpen(IAsyncResult result)
Microsoft.ServiceBus.Messaging.EventHubReceiver.OnEndOpen(IAsyncResult result)
Microsoft.ServiceBus.Messaging.ClientEntity.EndOpen(IAsyncResult result)
Microsoft.ServiceBus.Messaging.EventHubConsumerGroup.CreateReceiverAsyncResult.<>c.<GetAsyncSteps>b__15_1(CreateReceiverAsyncResult thisPtr, IAsyncResult r)
Microsoft.ServiceBus.Messaging.IteratorAsyncResult`1.StepCallback(IAsyncResult result)
Microsoft.ServiceBus.Common.AsyncResult.End[TAsyncResult](IAsyncResult result)
Microsoft.ServiceBus.Messaging.IteratorAsyncResult`1.RunSynchronously()
Microsoft.ServiceBus.Messaging.EventHubConsumerGroup.CreateReceiver(String partitionId, DateTime startingDateTimeUtc, Int64 epoch, ReceiverOptions receiverOptions)
Microsoft.Azure.Devices.E2ETests.MessageE2ETests.CreateEventHubReceiver(String deviceName, EventHubClient& eventHubClient) in D:\jenkins\workspace\csharp-windows-wip\e2e\Microsoft.Azure.Devices.E2ETests\MessageE2ETests.cs: line 606
Microsoft.Azure.Devices.E2ETests.MessageE2ETests.<SendMessageRecovery>d__71.MoveNext() in D:\jenkins\workspace\csharp-windows-wip\e2e\Microsoft.Azure.Devices.E2ETests\MessageE2ETests.cs: line 795
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.MessageE2ETests.<Message_AmqpConnectionLossSendRecovery_AmqpWs>d__26.MoveNext() in D:\jenkins\workspace\csharp-windows-wip\e2e\Microsoft.Azure.Devices.E2ETests\MessageE2ETests.cs: line 177
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
```

**Comment by jenoola** [comment]
Fixed in 1.20.0

**Comment by az-iot-builder-01** [comment]
@CIPop, @jenoola, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
